tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -949,11 +949,15 @@ function Use-AzureRmProfile
         }
       }
 
-      # Block user if they try to import a module to the session where a different version of the same module is already imported
-      if ($null -ne (Get-Module -Name $Module | Where-Object { $_.Version -ne $version} ))
+      # If a different profile's Azure Module was imported, block user
+      $importedModules = Get-Module "Azure*"
+      foreach ($importedModule in $importedModules) 
       {
-        Write-Error "A different version of module $module is already imported in this session. Start a new PowerShell session and retry the operation."
-        return
+        if (($null -ne $ProfileMap.$Profile.$($importedModule.Name)) -and ($ProfileMap.$Profile.$($importedModule.Name) -ne $importedModule.Version))
```

Comment:
Aren't the versions for a particular profile / module an array?  This seems to be treating them as a single value